File: MAP/lab6/src/models/expression/ConstExp.java
package models.expression;

import utils.IDictionary;
import utils.IHeap;

public class ConstExp implements IExpression {
    private int number;

    public ConstExp(int _number){
        number = _number;
    }

    @Override
    public int eval(IDictionary<String, Integer> d, IHeap<Integer, Integer> heap) {
        return number;
    }

    @Override
    public String toString(){
        return "" + number ;
    }
}

File: MAP/lab7/lab7/Models/ConstExp.cs
using lab7.Utils;

namespace lab7.Models
{
    public class ConstExp : IExpression
    {
        private int number;

        public ConstExp(int number)
        {
            this.number = number;
        }


        public int eval(IDictionary<string, int> dict)
        {
            return number;
        }

        public override string ToString()
        {
            return " "+ number;
        }
    }
}